xmtp_mls/src/groups/sync.rs

```diff
@@ -133,6 +134,28 @@ impl MlsGroup {
         Ok(())
     }
 
+    pub(super) async fn sync_until_all_intents_resolved<ApiClient>(
```

Comment:
Might be more accurate to name this `sync_until_last_jntent_resolved`. Earlier intents can still fail and this will succeed.